doc/memos/rdm3.md

```diff
@@ -0,0 +1,98 @@
+RIOT Maintainers Community
+Developer Memo: 3
+Author(s): Martine Lenders
+Date: January 2017
+
+Network programming guide
+=========================
+
+## Abstract
+This memo describes and justifies the usage and existence of RIOT's various
+networking APIs.
+
+
+## Status
+This document is a product of the community of RIOT maintainers, and aims to 
+represent the consensus within this community.
+The content of this document is licensed with a Creative Commons CC-BY-SA 
+license.
+
+## Terminology
+This memo uses the following terminology and acronyms:
+
+- network stack: the architecture to implement networking
+- GNRC: RIOT's default network stack
+- netapi: GNRC's inter-modular communication API
+- IPC: Inter-process communication
+- sock: RIOT's primary API for applications to the network stack
```

Comment:
"applications to the network stack" sounds weird to me. Maybe just "RIOT's primary API for network applications"? Same for posix two lines down.